mathesar/imports/excel.py

```diff
@@ -50,9 +50,16 @@ def remove_empty_rows_and_columns_from_dataframe(df):
 def create_db_table_from_excel_data_file(data_file, name, schema, comment=None):
     db_name = schema.database.name
     engine = create_mathesar_engine(db_name)
-    dataframe = remove_empty_rows_and_columns_from_dataframe(
-        pandas.read_excel(data_file.file.path)
-    )
+    header_row = 0 if data_file.header else None
+    try:
+        dataframe = remove_empty_rows_and_columns_from_dataframe(
+            pandas.read_excel(data_file.file.path, data_file.sheet_name, header=header_row)
+        )
+    except ValueError:
+        # If sheet number has been passed as a parameter instead of sheet name.
+        dataframe = remove_empty_rows_and_columns_from_dataframe(
+            pandas.read_excel(data_file.file.path, int(data_file.sheet_name), header=header_row)
+        )
```

Comment:
I am a bit skeptical about this, I think we should _only_ use the index of the file so that our django model could stay consistent, this could be achieved by searching the name of the excel sheet in the list of all the sheet_names and then finding its index. However, finding the list of sheet_names could cause a lot of overhead using the `ExcelFile.sheet_names` or `df.keys()` as they require to load the whole excel file in memory instead of just the specified sheet as done here. I did however find a [method](https://stackoverflow.com/a/56320206) to efficiently extract sheet names but idk if its worth it. I'd like to know your thoughts on this @dmos62.

Reply:
I notice there's some confusion between index and name here. Sometimes the `sheet_name` is cast to an `int`, which I presume implies that it acts as an index. `sheet_index` and `sheet_name` should be distinct and if we'd want to use some kind of polymorphic variable that's sometimes a name and sometimes an index, we'd probably name it `sheet_id` (since both a name and an index are identifiers), but I don't like the idea of a variable sometimes being one thing and sometimes something else.

For what it's worth, I think an index identifier has a better simplicity/usefulness relation than a name identifier. We could support both, but keeping it minimal is desirable right now, I think.

The SO answer @Anish9901 linked to mentioned that loading excel files gets exponentially slower with size (in Pandas). Supposing that's true, we'd do well to fix that at some point. Maybe now. If not, we should definitely put a `TODO` comment with a link to that SO thread explaining what the problem is and what the solution could be (where the relevant code is found).